ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/optimisation/fillers/MaxLoopFlowFiller.java

```diff
@@ -48,9 +48,14 @@ public void update(LinearRaoData linearRaoData, LinearRaoProblem linearRaoProble
      * we define loopFlowShift = PTDF * NetPosition, then
      * -maxLoopFlow + loopFlowShift <= flowVariable <= maxLoopFlow + loopFlowShift,
      */
-    private void buildMaxLoopFlowConstraint(LinearRaoData linearRaoData, LinearRaoProblem linearRaoProblem) {
-        LoopFlowComputation loopFlowComputation = new LoopFlowComputation(linearRaoData.getCrac()); //todo optimization => do not recalculate loopflowShifts, use the one in the CracResult
-        Map<Cnec, Double> loopFlowShifts = loopFlowComputation.buildZeroBalanceFlowShift(linearRaoData.getNetwork()); //todo: validate assumption that "loopflowShift" does not change "much"
+    private void buildMaxLoopFlowConstraint(LinearRaoData linearRaoData, LinearRaoProblem linearRaoProblem, LinearRaoParameters linearRaoParameters) {
+        Map<Cnec, Double> loopFlowShifts;
+        LoopFlowComputation loopFlowComputation = new LoopFlowComputation(linearRaoData.getCrac());
+        if (!Objects.isNull(linearRaoParameters.getExtendable()) && linearRaoParameters.getExtendable().isLoopflowApproximation()) {
+            loopFlowShifts = loopFlowComputation.buildLoopflowShiftsApproximation(linearRaoData.getCrac());
+        } else {
+            loopFlowShifts = loopFlowComputation.buildZeroBalanceFlowShift(linearRaoData.getNetwork());
+        }
```

Comment:
I'm not sure of what I say, but should'nt we always use the previously stored shifts here, as : 

* if we have` "loopflow-approximation" : false`, we want to be quick and not rerun a loopflow computation
* if we have` "loopflow-approximation" : true`, the loopflow have already been computed on the same network in the SensitivityAnalysisEngine (either during the first sensi analysis of the LinearRao, either during the sensi of the previous iteration of the LinearRao)

What do you think ? As I understand it we compute 2 times the loopflow per iteration of the LinearRao (one here in the filler, and one in the sensiEngine) and we are doing 2 times the same calculation.

Reply:
"loopflow-approximation = false" means that we compute the exact loopflow by re-running a sensi for PTDF; " = true" means that we use previous calculated PTDF * netposition value (as mentioned in US7.8).

Reply:
Could you please try to always go in the if part of this if/else and see if it changes the results on some cases where  `"loopflow-approximation" : false`. I am pretty sure that the shifts have already been calculated on the same network here and that we do not need to go for the "accurate" loop-flow computation.

Reply:
It might not be the case for the first iteration with the current design though... let's wait for the next meeting on this topic.